pkg/apis/pipeline/v1/pipeline_types.go

```diff
@@ -267,6 +267,23 @@ func (pt *PipelineTask) IsMatrixed() bool {
 	return pt.Matrix != nil && (pt.Matrix.hasParams() || pt.Matrix.hasInclude())
 }
 
+// extractAllParams extracts all the parameters in a PipelineTask:
+// - pt.Params
+// - pt.Matrix.Params
+// - pt.Matrix.Include.Params
+func (pt *PipelineTask) extractAllParams() Params {
+	allParams := pt.Params
```

Comment:
will this end up modifying `pt.Params`?

Reply:
@lbernick No, it does not modify `pt.Params`.